projects/packages/publicize/src/class-publicize-base.php

```diff
@@ -1606,6 +1606,65 @@ public function get_social_opengraph_image( $post_id ) {
 		return array();
 	}
 
+	/**
+	 * Returns the image size in bytes of a remote image.
+	 *
+	 * @param  string $image_url       Image URL.
+	 * @return integer|null $bytes      Image size in bytes, or null if request failed.
+	 */
+	public function get_remote_filesize( $image_url ) {
+		$response = wp_remote_get( $image_url, array( 'method' => 'HEAD' ) );
+		$size     = wp_remote_retrieve_header( $response, 'content-length' );
+
+		return $size ? $size : null;
+	}
+
+	/**
+	 * Reduce the filesize of an image by reducing the dimensions. Uses Photon.
+	 * Returns null if the image cannot be reduced enough.
+	 *
+	 * @param string $url Image URL.
+	 * @param int    $width Image width.
+	 * @param int    $height Image height.
+	 * @param int    $max_filesize Maximum filesize in bytes. Default is 4MB.
+	 * @param int    $tries Number of times to try reducing the image size. Default is 5.
+	 * @return array|null
+	 */
+	public function reduce_file_size( $url, $width, $height, $max_filesize = 4000000, $tries = 5 ) {
+		$filesize = $this->get_remote_filesize( $url );
+		// If we cannot get the size, or it's small enough, we bail.
+		if ( empty( $filesize ) || $filesize <= $max_filesize ) {
+			return null;
+		}
+
+		while ( $tries > 0 && $filesize > $max_filesize ) {
+			$width   *= 0.75;
+			$height  *= 0.75;
+			$url      = jetpack_photon_url(
+				$url,
+				array(
+					'w' => $width,
+					'h' => $height,
+				)
+			);
+			$filesize = $this->get_remote_filesize( $url );
+			--$tries;
+		}
+
+		// If the image is still too large, we failed.
+		if ( $filesize > $max_filesize ) {
+			// Track this to see if blank images in shared posts may be due to this code path.
+			bump_stats_extras( 'publicize_error_condition', 'overlarge-image-' . wpcom_blog_site_id_label() );
```

Comment:
Is this available on the Jetpack site?

Reply:
Good point, it's only on WPCOM side. I removed it for now